src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.StandardProperties.cs

```diff
@@ -0,0 +1,302 @@
+        /// </summary>
+        public CngAlgorithm Algorithm
+        {
+            get
+            {
+                string algorithm = _keyHandle.GetPropertyAsString(KeyPropertyName.Algorithm, CngPropertyOptions.None);
+                // Desktop compat: Don't check for null. Just let CngAlgorithm handle it.
+                return new CngAlgorithm(algorithm);
+            }
+
+        }
+
+        /// <summary>
+        ///     Name of the algorithm this key can be used with
+        /// </summary>
+        public CngAlgorithmGroup AlgorithmGroup
+
+        {
+            get
+            {
+                string algorithmGroup = _keyHandle.GetPropertyAsString(KeyPropertyName.AlgorithmGroup, CngPropertyOptions.None);
+                if (algorithmGroup == null)
+                    return null;
+                return new CngAlgorithmGroup(algorithmGroup);
+            }
+        }
+
+        /// <summary>
+        ///     Export restrictions on the key
+        /// </summary>
+        public CngExportPolicies ExportPolicy
+        {
+            get
+            {
+                CngExportPolicies policy = (CngExportPolicies)_keyHandle.GetPropertyAsDword(KeyPropertyName.ExportPolicy, CngPropertyOptions.None);
+                return policy;
+            }
+        }
+
+        /// <summary>
+        ///     Native handle for the key
+        /// </summary>
+        public SafeNCryptKeyHandle Handle
+        {
+            get
+            {
+                return _keyHandle.Duplicate();
+            }
+        }
+
+        /// <summary>
+        ///     Is this key ephemeral or persisted
+        /// </summary>
+        /// <remarks>
+        ///     Any ephemeral key created by the CLR will have a property 'CLR IsEphemeral' which consists
+        ///     of a single byte containing the value 1. We cannot detect ephemeral keys created by other
+        ///     APIs and imported via handle.
+        /// </remarks>
+        public bool IsEphemeral
+        {
+            get
+            {
+                unsafe
+                {
+                    byte propertyValue;
+                    int cbResult;
+                    ErrorCode errorCode = Interop.NCrypt.NCryptGetProperty(_keyHandle, KeyPropertyName.ClrIsEphemeral, &propertyValue, sizeof(byte), out cbResult, CngPropertyOptions.CustomProperty);
+                    if (errorCode != ErrorCode.ERROR_SUCCESS)
+                    {
+                        // Third party Key providers, and Windows PCP KSP won't recognize this property;
+                        // and Win32 layer does not enforce error return contract.
+                        // Therefore, they can return whatever error code they think appropriate.
+                        return false;
+                    }
+
+                    if (cbResult != sizeof(byte))
+                        return false;
+
+                    if (propertyValue != 1)
+                        return false;
```

Comment:
We can’t Debug.Assert the value since an app could set the “CLR IsEphemeral” property with an illegal value himself (ah, the joys of in-band signaling.) So an illegal value isn’t
necessarily a sign of a bug inside CoreFx. The desktop and CoreFx code use the same logic – they both return “IsEphemeral = false” for 2 and other illegal values. I just verified this on the desktop and CoreFx.